yarpl/src/yarpl/observable/Subscriptions.cpp

```diff
@@ -13,7 +13,10 @@ namespace observable {
  */
 void Subscription::cancel() {
   cancelled_ = true;
-  for(auto& subscription : *tiedSubscriptions_.rlock()) {
+  // Lock must be obtained here and not in the range expression for it to
+  // apply to the loop body.
+  auto locked = tiedSubscriptions_.rlock();
```

Comment:
needs exclusive access; `wlock()`

Reply:
Actually I am not sure if we want the lock to be applied to the loop body at all. 
Namely the only reason for locking might be to access to the list in a safe way, but it doesn't mean that we would like to make the access to the subscription be synchronized. 

'subscription' in here is already a Reference, so it will not get destroyed or else.

So I don't believe this change is necessary..